time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java

```diff
@@ -152,18 +169,48 @@ static String checkString(String token) {
         return token.isEmpty() ? null : token;
     }
 
+    class CsvParserConfig {
+        boolean versioned;
+        TimeFormat timeFormat;
+
+        CsvParserConfig() {
+            this(true, TimeFormat.DATE_TIME);
+        }
+
+        public CsvParserConfig(boolean versioned, TimeFormat timeFormat) {
+            this.versioned = versioned;
+            this.timeFormat = timeFormat;
+        }
+
+        public boolean versioned() {
+            return versioned;
+        }
+
+        public TimeFormat timeFormat() {
+            return timeFormat;
+        }
+    }
+
     class CsvParsingContext {
         final List<String> names;
+        final CsvParserConfig csvParserConfig;
+        final int fixedColumns;
 
         final TimeSeriesDataType[] dataTypes;
         final Object[] values;
 
-        final List<ZonedDateTime> times = new ArrayList<>();
+        final List<Long> times = new ArrayList<>();
 
         TimeSeriesIndex refIndex;
```

Comment:
Same comment there: you could also fix the declaration of the class and the fields

Reply:
Done